### ISSUE TITLE
TF always reports pending changes

### ISSUE THREAD

**Issue by unicolet** [checkly/terraform-provider-checkly#23]
**Describe the bug**

API checks keep report pending changes, even though no changes have been made to the definition.

***Version***

0.6.3 with terraform 0.11. Terraform state is on S3 bucket.

```
Terraform v0.11.14
.
└── provider.checkly >= 0.6.3
```

**To Reproduce**

create an API check:

```
resource "checkly_check" "google" {
  name                   = "google"
  type                   = "API"
  activated              = true
  should_fail            = false
  frequency              = 1
  degraded_response_time = 3000
  max_response_time      = 6000
  tags = [
    "testing",
    "bug"
  ]

  locations = [
    "eu-central-1"
  ]

  request {
    follow_redirects = false
    url              = "https://www.google.com"
    
    basic_auth {
        username = ""
        password = ""
    }

    assertion {
      source     = "STATUS_CODE"
      property   = ""
      comparison = "EQUALS"
      target     = "200"
    }
  }
}
```

run terraform plan, apply. Then run terraform plan again and this is reported [slightly edited]:

```
Terraform will perform the following actions:                                                           
                                                                                                        
  ~ checkly_check.google                                                                                    
      request.1080913471.assertion.#:                                                       "1" => "0" 
      request.1080913471.assertion.3474411808.comparison:                                   "EQUALS" => ""  
      request.1080913471.assertion.3474411808.property:                                     "" => ""                                                     
      request.1080913471.assertion.3474411808.source:                                       "STATUS_CODE" => ""
      request.1080913471.assertion.3474411808.target:                                       "200" => ""   
      request.1080913471.body:                                                              "" => ""
      request.1080913471.body_type:                                                         "NONE" => ""       
      request.1080913471.follow_redirects:                                                  "false" => "false"
      request.1080913471.method:                                                            "GET" => ""
      request.1080913471.url:                                                               "https://www.google.com" => ""
      request.1800643113.assertion.#:                                                       "0" => "1"       
      request.1800643113.assertion.3474411808.comparison:                                   "" => "EQUALS"
      request.1800643113.assertion.3474411808.property:                                     "" => ""                                                     
      request.1800643113.assertion.3474411808.source:                                       "" => "STATUS_CODE"
      request.1800643113.assertion.3474411808.target:                                       "" => "200"
      request.1800643113.basic_auth.#:                                                      "0" => "1"       
      request.1800643113.basic_auth.3546472845.password:                                    "" => ""  
      request.1800643113.basic_auth.3546472845.username:                                    "" => ""  
      request.1800643113.body:                                                              "" => ""  
      request.1800643113.body_type:                                                         "" => "NONE"
      request.1800643113.follow_redirects:                                                  "" => "false"
      request.1800643113.headers.%:                                                         "" => <computed>
      request.1800643113.method:                                                            "" => "GET"
      request.1800643113.query_parameters.%:                                                "" => <computed> 
      request.1800643113.url:                                                               "" => "https://www.google.com"
```

**Expected behavior**

TF should not have reported changes after the configuration had been just applied.

**TF Resource files**

```sh
# cat main.tf 
variable "checkly_api_key" {}

provider "checkly" {
  version = ">= 0.6.3"
  api_key = "${var.checkly_api_key}"
}
```
**Additional context**

**Comment by tnolet** [comment]
@unicolet thanks for reporting. I can clearly see the plan churn happening on the nested `request` object.

**Comment by tnolet** [comment]
@unicolet @kjda I can see that this is happening again due to our "friend" the `basic_auth` setting.

If you remove this stanza from the `request` resource, you should see no more churn:

```
    basic_auth {
        username = ""
        password = ""
    }
```

However, we should also accept it and not create churn in the first place

**Comment by unicolet** [comment]
@tnolet thanks for the quick feedback :heart: 

I can confirm that removing the `basic_auth` stanza from the check got rid of the churn _for the check I reported in this issue_. I have other 2 (the real ones, not one I created for the purpose of reproducing this issue) where the churn will not go away, even after applying the plan once with the removed `basic_auth` stanza :cry: 

If that's helpful those 2 were created with version `0.6.2` of the provider and the output reported by tf plan is:

```
  ~ checkly_check.tf_mycheckname
      alert_settings.2431968856.escalation_type:                                            "RUN_BASED" => ""
      alert_settings.2431968856.reminders.#:                                                "1" => "0"
      alert_settings.2431968856.reminders.2388976591.amount:                                "0" => "0"
      alert_settings.2431968856.reminders.2388976591.interval:                              "5" => "0"
      alert_settings.2431968856.run_based_escalation.#:                                     "1" => "0"
      alert_settings.2431968856.run_based_escalation.633714007.failed_run_threshold:        "1" => "0"
      alert_settings.2431968856.ssl_certificates.#:                                         "1" => "0"
      alert_settings.2431968856.ssl_certificates.692866002.alert_threshold:                 "30" => "0"
      alert_settings.2431968856.ssl_certificates.692866002.enabled:                         "true" => "false"
      alert_settings.2431968856.time_based_escalation.#:                                    "1" => "0"
      alert_settings.2431968856.time_based_escalation.3327931791.minutes_failing_threshold: "5" => "0"
      alert_settings.3439023353.escalation_type:                                            "" => ""
      alert_settings.3439023353.reminders.#:                                                "0" => "0"
      alert_settings.3439023353.run_based_escalation.#:                                     "0" => "0"
      alert_settings.3439023353.ssl_certificates.#:                                         "0" => "1"
      alert_settings.3439023353.ssl_certificates.692866002.alert_threshold:                 "" => "30"
      alert_settings.3439023353.ssl_certificates.692866002.enabled:                         "" => "true"
      alert_settings.3439023353.time_based_escalation.#:                                    "0" => "0"
```

**Comment by kjda** [comment]
I'm investigating this hoping to come up with a fix asap

**Comment by unicolet** [comment]
thanks, I'll check this ASAP
